`pyvista_ndarray` leaks a `vtkWeakReference` through `point_data.__setitem__`

## Summary

Assigning a `pyvista_ndarray` subclass (e.g. the result of slicing `mesh.points`) into `mesh.point_data[...]` leaks one `vtkWeakReference` per call. The leak survives the owning mesh, so `check_gc` in `tests/plotting/conftest.py` flags a new VTK object at fixture teardown.

The leak reproduces on both Python 3.13 and 3.14 in isolation, but `check_gc` only fails on 3.14. CPython 3.14 finalizes numpy subclasses with `__array_finalize__` differently, so the chain below stays reachable from `gc.get_objects()` at teardown on 3.14 and gets collected on 3.13.

I uncovered this in #8498 and had to change my tests to avoid this:  `vtk==9.6.1`, `numpy==2.4.4`, `python==3.14.3`

## pytest reproducer using the existing `check_gc` fixture

```python
# tests/plotting/test_gc_leak.py
import pyvista as pv

def test_point_data_assignment_leaks_vtk_weak_reference():
    mesh = pv.Sphere()
    mesh.point_data['data'] = mesh.points[:, 2].astype(float)
```

On Python 3.14 this fails with:

```
AssertionError: Not all objects GCed:
  vtkWeakReference at 0x...: ['pyvista_ndarray([...])']
assert 1 == 0
```

A plain `np.ndarray` avoids the leak:

```python
import numpy as np
mesh.point_data['data'] = np.array(mesh.points[:, 2].astype(float))
```

## Reference chain

```
vtkDataArray  (stored on mesh.point_data)
  -> _numpy_reference  (set by numpy_to_vtk to keep the buffer alive)
       -> pyvista_ndarray  (allocated in
                            pyvista/core/datasetattributes.py:822
                            `copy = pyvista_ndarray(data)`)
            -> .dataset = vtkWeakReference()
               (set unconditionally in pyvista_ndarray.__new__ line 83)
```

When the mesh is deleted, Python 3.14's GC does not break this chain, so the `vtkWeakReference` stays reachable.

## Proposed fix?

Skip allocating a `vtkWeakReference` in `pyvista_ndarray.__new__` when the caller passes `dataset=None`:

```python
# pyvista/core/pyvista_ndarray.py, __new__
obj.association = association
if dataset is None:
    obj.dataset = None
else:
    obj.dataset = _vtk.vtkWeakReference()
    if isinstance(dataset, _vtk.VTKObjectWrapper):
        obj.dataset.Set(dataset.VTKObject)
    else:
        obj.dataset.Set(cast('_vtk.vtkDataSet', dataset))
```

--- a/pyvista/core/pyvista_ndarray.py
+++ b/pyvista/core/pyvista_ndarray.py
@@ -80,11 +80,14 @@ class pyvista_ndarray(_NoNewAttrMixin, np.ndarray):  # numpydoc ignore=PR02  # n
             raise TypeError(msg)
 
         obj.association = association
-        obj.dataset = _vtk.vtkWeakReference()
-        if isinstance(dataset, _vtk.VTKObjectWrapper):
-            obj.dataset.Set(dataset.VTKObject)
+        if dataset is None:
+            obj.dataset = None
         else:
-            obj.dataset.Set(cast('_vtk.vtkDataSet', dataset))
+            obj.dataset = _vtk.vtkWeakReference()
+            if isinstance(dataset, _vtk.VTKObjectWrapper):
+                obj.dataset.Set(dataset.VTKObject)
+            else:
+                obj.dataset.Set(cast('_vtk.vtkDataSet', dataset))
         return obj
 
     def __array_finalize__(self: pyvista_ndarray, obj: npt.NDArray[Any] | None) -> None:

--- a/tests/core/test_pyvista_ndarray.py
+++ b/tests/core/test_pyvista_ndarray.py
@@ -1,5 +1,6 @@
 from __future__ import annotations
 
+import gc
 import re
 from unittest import mock
 
@@ -7,6 +8,7 @@ import numpy as np
 import pandas as pd
 import pytest
 
+import pyvista as pv
 from pyvista import examples
 from pyvista import pyvista_ndarray
 from pyvista import vtk_points
@@ -116,3 +118,20 @@ def test_wrap_pandas(obj_in):
     array = pyvista_ndarray(obj_in)
     df = pd.DataFrame(array)
     assert np.shares_memory(df.values, array)
+
+
+def test_no_dataset_does_not_allocate_weak_reference():
+    # Regression test for https://github.com/pyvista/pyvista/issues/8532
+    arr = pyvista_ndarray([1.0, 2.0, 3.0])
+    assert arr.dataset is None
+
+
+def test_point_data_assignment_does_not_leak_vtk_weak_reference():
+    # Regression test for https://github.com/pyvista/pyvista/issues/8532
+    mesh = pv.Sphere()
+    mesh.point_data['data'] = mesh.points[:, 2].astype(float)
+    del mesh
+    gc.collect()
+
+    leaked = [o for o in gc.get_objects() if isinstance(o, _vtk.vtkWeakReference)]
+    assert leaked == []

--- a/tests/plotting/test_picking.py
+++ b/tests/plotting/test_picking.py
@@ -606,7 +606,6 @@ def test_block_picking(multiblock_poly):
 
 
 @pytest.mark.parametrize('mode', ['mesh', 'cell', 'face', 'edge', 'point'])
-@pytest.mark.skip_check_gc("vtkDataArray not gc'd with 'point' mode on Python 3.14 vtk dev wheels")
 def test_element_picking(mode):
     class Tracker:
         def __init__(self):

--- a/tests/plotting/test_plotting.py
+++ b/tests/plotting/test_plotting.py
@@ -2104,7 +2104,6 @@ def test_volume_rendering_from_plotter(uniform):
 
 
 @skip_windows_mesa  # due to opacity
-@pytest.mark.skip_check_gc("vtkWeakReference not gc'd on Python 3.14 vtk dev wheels")
 def test_volume_rendering_rectilinear(uniform):
     grid = uniform.cast_to_rectilinear_grid()
 
@@ -3828,7 +3827,7 @@ def test_plotter_lookup_table(sphere, verify_image_cache):
 
 
 @skip_windows_mesa  # due to opacity
-@pytest.mark.skip_check_gc("vtkTypeUInt8Array not gc'd on Python 3.14 vtk dev wheels")
+@pytest.mark.skip_check_gc("vtkTypeUInt8Array not gc'd on Python 3.14")
 def test_plotter_volume_lookup_table(uniform):
     uniform.set_active_scalars('Spatial Point Data')
 

